src/partition/ReceivedPartition.hpp

```diff
@@ -104,8 +99,28 @@ class ReceivedPartition : public Partition {
   /// Min global vertex IDs of remote connected ranks
   std::vector<int> _remoteMinGlobalVertexIDs;
 
-  template <typename T>
-  friend void PartitionTests::ReceivedPartitionTests::testParallelSetOwnerInformation(T &mesh, int dimensions);
+  // Make the fixture friend of this class
+  friend struct receivedPartitionFixture;
+};
+
+/*
+ * @brief A fixture that is used to access private functions of the receivedPartition class.
+ *
+ * The fixture can be used to call private functions for individual testing. 
+ */
+struct receivedPartitionFixture {
+  void createOwnerInformation(ReceivedPartition &part)
+  {
+    part.createOwnerInformation();
+  }
+  void tagMeshFirstRound(ReceivedPartition &part)
+  {
+    part.tagMeshFirstRound();
+  }
+  void prepareBoundingBox(ReceivedPartition &part)
+  {
+    part.prepareBoundingBox();
+  }
```

Comment:
Moving this into a dedicated testing header would make things cleaner. Maybe `src/partition/tests/fixtures.hpp`?

Reply:
Done.